wfe/wfe.go

```diff
@@ -2079,6 +2089,17 @@ func (wfe *WebFrontEndImpl) Authz(
 				response)
 			return
 		}
+
+		if authz.Status == acme.StatusPending {
+			// Check for the existence of a challenge which state is processing, and add the
+			// Retry-After header if there is one.
+			for _, c := range authz.Challenges {
+				if c.Status == acme.StatusProcessing {
```

Comment:
I think it's likely that we need to do a `c.RLock()` and `c.RUnlock()` around the access to `c.Status`, or it may be a race.

Reply:
@alexzorin 
Thank you for your comment. I will attempt to fix this problem.

Reply:
This problem was fixed by commit [6f709f9](https://github.com/letsencrypt/pebble/pull/380/commits/6f709f914d48e7bc067683ddcc851c3701c15744)